### PR TITLE
Add user customization support

### PR DESCRIPTION
**Pull request by agoelzer**
allows users to customize their user experience - in particular:
- add/remove view columns
- add/remove buttons
- change column titles
The user customization is being stored in the browser's localStorage

Note: I transitioned the customizations.js file (java script with call back support) into a customizations.json file, so the data can be easily validated (different PR) and avoid security exposure by not allowing to add arbitrary JavaScript code by the end-user. Since the JavaScript `eval()` function also exposes a security risk, only needed functionality is implemented in it's own parser for the `value` and `visible` properties.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,10 +102,10 @@ start-dev: setup-integration-tests ## launch WebUI/ControlPlane/Proxy for develo
 
 .PHONY: stop-dev
 stop-dev: teardown-integration-tests ## stop development environment processes (WebUI/ControlPlane/Proxy)
-	@PID=$(shell netstat -a -n -p 2> /dev/null | sed -n -E "s/.* 0\.0\.0\.0:3000 .* LISTEN .* ([0-9]+)\/node/\1/p")
-	@if [ "${PID}" != "" ] ; then kill -9 ${PID}; fi
-	@PID=$(shell docker ps -aq --filter "name=nuodb-webui-dev")
-	@if [ "${PID}" != "" ] ; then docker stop ${PID}; fi
+	@PID=$(shell netstat -a -n -p 2> /dev/null | sed -n -E "s/.* 0\.0\.0\.0:3000 .* LISTEN .* ([0-9]+)\/node/\1/p"); \
```
Comment (adriansuarez) [inline Makefile:105]: What is this scraping?

File: ui/customizations.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # NuoDBaaS WebUI customizations
 
 The NuoDBaaS WebUI allows for UI customizations to handle special cases which are not defined in the OpenAPI spec file. These are currently action buttons in the views and in the future will be extended to simplified forms.
-The configuration file is currently stored in `public/customizations.js` and is a JavaScript file containing configurations, function callbacks and constants referencing other parts of the file to avoid duplication.
+The configuration file is currently stored in `public/customizations.json` and is a JSON file containing configurations, macros ("value" and "disabled" attributes.
```
Comment (adriansuarez) [inline ui/customizations.md:4]: Missing closing parenthesis?

File: ui/customizations.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ function getCustomizations() {
     "views": {
         "/databases": {
             status: {
-                value: (data) => data.status && (data.status.ready ? "Ready" : "Not Ready"),
+                value: "status.ready),
```
Comment (adriansuarez) [inline ui/customizations.md:14]: Should the closing parenthesis be replaced by a quote (i.e. `value: "status.ready",`

File: Makefile (hunk 1 above)
Comment (agoelzer) [reply Makefile:105]: it is finding the PID of the `node` process which is listening on port 3000 on all IP's and kills the process - this process is started by the "run-dev" target and is responsible to run the React App in development mode. The other process which is killed a few lines above is the reverse proxy to workaround CORS issues with the NuoDB Control Plane.

**Review by kontaras — APPROVED** [review]
Looks good except for a small cosmetic issue.

File: ui/customizations.md (hunk 2 above)
Comment (agoelzer) [reply ui/customizations.md:4]: Done

File: ui/customizations.md (hunk 3 above)
Comment (agoelzer) [reply ui/customizations.md:14]: Done

[the remark this answers is not in the document]
Comment (agoelzer) [reply ui/customizations.md:14]: Done

**Pull request merged**